src/name_core.jl

```diff
@@ -102,15 +100,16 @@ function permute_dimnames(dimnames::NTuple{N, Symbol}, perm) where N
     return compile_time_return_hack(new_dimnames)
 end
 
-"""
-    default_inds(dimnames::Tuple)
-This is the default value for all indexing expressions using the given dimnames.
-Which is to say: take a full slice on everything
-"""
-function default_inds(dimnames::NTuple{N}) where N
-    # 0-Allocations
-    values = ntuple(_->Colon(), N)
-    return NamedTuple{dimnames, NTuple{N, Colon}}(values)
+
+Base.@pure function tuple_issubset(lhs::Tuple{Vararg{Symbol,N}}, rhs::Tuple{Vararg{Symbol,M}}) where {N,M}
```

Comment:
Why do we need this? What is the behaviour that's not available with `issubset`? (and does the answer to that warrant a comment here?)

```julia
julia> issubset((:a, :b), (:a, :b, :c))
true

julia> issubset((:a, :b, :c), (:a, :b))
false

julia> issubset((:a, :b), (:a, :b))
true
```
versus
```julia
julia> tuple_issubset((:a, :b), (:a, :b, :c))
true

julia> tuple_issubset((:a, :b, :c), (:a, :b))
false

julia> tuple_issubset((:a, :b), (:a, :b))
true
```

Reply:
A good question.
The answer is: this one is tagged `@pure`
and it is a fairly safe use of `@pure`.
(According to discussion I had in JuliaSlack #internal yesterday)
The purity means that we get to constant propegate

Reply:
please can you add a comment to the code explaining that?